content/en/docs/features/monitoring-and-diagnostics/monitoring-device-connectivity.md

```diff
@@ -0,0 +1,89 @@
+
+   ```golang
+   now := time.Now().UnixNano()
+   period := N * time.Minute
+   recordID = now - (now % int64(period))
+   ```
+
+3. **Shared Record Updating**: Regularly, each CoAP gateway instance contributes to a shared record among CoAP gateways in the database by appending its unique ID. The record being updated changes cyclically over time. Both the current shared record and the previous one are retained in the database for historical reference. The structure of the record is defined by the `SharedRecord` type in Golang:
+
+   ```golang
+   type SharedRecord struct {
+       RecordID          int64    `bson:"_id"`            // ID of the record moving forward in time
+       CoAPGatewayIDs    []string `bson:"coapgatewayids"` // IDs of lived CoAP gateways during the period
+   }
+   ```
+
+   If the record has already been created, move on to the subsequent step. Otherwise, await the next time interval and initiate the process again starting from step `2.`
+
+4. **Termination Detection**: Upon the creation of a new record, the system examines records with a `recordID` corresponding to the time within the last periods. This examination encompasses CoAP gateways that were operational before the current period.
+
+5. **Gateway Exclusion**: CoAP gateways active within the current period and the previous period are excluded from the list of operational gateways derived in the previous step.
+
+6. **Device Transition to Offline**: The remaining CoAP gateways in the list are identified as terminated. To ensure accurate device status, the system uses the Resource Aggregate command `TerminateConnectionsByService`. This command transitions devices associated with terminated CoAP gateways to an offline state.
+
+   {{< warning >}}
+
+   The `TerminateConnectionsByService` function is designed exclusively for service usage. As a result, this command does not rely on JWT for authorization. It's important to note that this function should not be callable from public endpoints. Instead, the sole method of authorization employed is mutual TLS authentication between the CoAP gateway and the Resource Aggregate.
+
+   {{< /warning >}}
+
+7. **Record Cleanup**: For effective management, records corresponding to periods before the last two update periods are purged. This practice maintains relevant records and prevents unnecessary data accumulation.
+
+In summary, the process of managing device states and setting them to an offline state within the context of CoAP gateways involves the systematic updating of shared records with unique gateway IDs, cyclic record management, dynamic `recordID` calculation, detection of terminated gateways, transition of associated devices to an offline state, and routine record cleanup. This comprehensive approach ensures that the documentation accurately reflects the current operational status of devices and gateways while maintaining historical context.
+
+### Example
```

Comment:
Good example, illustrates the algorithm well!

Reply:
:)